### DISCUSSION TITLE
Vite builds the lib package, and react-dom.development.js is built into it

### DISCUSSION THREAD

**Discussion by TomasHawKing** [vitejs/vite-plugin-react#209] [Q&A]
When I build in production mode, the build product should theoretically not contain the react-dom.development.js package, but it is currently found that the build result does contain react-dom.development.js

**visualizer result：**
![image](https://github.com/vitejs/vite-plugin-react/assets/37299235/ca66ed17-7e66-4e25-84b9-94d42a53b082)

**vite.lib.config.ts:**
```
export default ({ mode }) => {
  const analyzer = { ...loadEnv(mode, process.cwd()) };
  const analyzerPlugin = analyzer.VITE_ANALYZER
    ? [visualizer({ open: true, gzipSize: true, brotliSize: true, template: 'treemap' }) as PluginOption]
    : [];

  return defineConfig({
    build: {
      lib: {
        entry: './lib/index.tsx',
        name: 'xxxx',
        fileName(format) {
          return `login-pass.${format}.js`;
        },
        formats: ['es', 'umd'],
      },
      target: 'es2015',
      rollupOptions: {
        plugins: [...analyzerPlugin, terser()],
        output: {
          // Replace process.env.NODE_ENV with ‘production’ value in UMD package
          intro: `const process = { env: { NODE_ENV: JSON.stringify('production')} }`,
        },
      },
    },

    resolve: {
      alias: {
        '@': path.resolve(__dirname, 'lib'),
      },
    },
  });

```

**Comment by ArnaudBarre** [comment]
In 95% of the cases, react should be set to external for building libs and added to the peer dependencies of your lib.
In other more complex cases, you should you define `process.env.NODE_ENV` to `"production"` so that tree shaking can do it's job, your current setup only define it for runtime not build time. But again this should only be done in some very particular advanced usecases
